components/layout/query.rs

```diff
@@ -125,6 +125,68 @@ pub fn process_client_rect_request(node: ServoThreadSafeLayoutNode<'_>) -> Rect<
         .unwrap_or_default()
 }
 
+/// Process a query for the current CSS zoom of an element.
+/// <https://drafts.csswg.org/cssom-view/#dom-element-currentcsszoom>
+///
+/// Returns the effective zoom of the element, which is the product of all zoom
+/// values from the element up to the root. Returns 1.0 if the element is not
+/// being rendered (has no associated box or has display: none).
+pub fn process_current_css_zoom_query(node: ServoLayoutNode<'_>) -> f32 {
+    // If the element has no layout data (not being rendered), return 1.0
+    let Some(layout_data) = node.to_threadsafe().inner_layout_data() else {
+        return 1.0;
+    };
+    let layout_box = layout_data.self_box.borrow();
+    let Some(layout_box) = layout_box.as_ref() else {
+        return 1.0;
+    };
+
+    // Get the element's style to check if it has display: none
+    let Some(base_style) = layout_box.with_first_base(|base| base.style.clone()) else {
+        return 1.0;
+    };
+
+    // If element has display: none, return 1.0 according to spec
```

Comment:
This seems unnecessary, because with `display: none` there is no box so it has been handled above